### ISSUE TITLE
large prune fails

### ISSUE THREAD

**Issue by wscott** [restic/restic#725]
## Output of `restic version`
restic 0.3.3 (v0.3.3-4-g9f1f174)
compiled with go1.7.4 on linux/amd64

## Expected behavior
I ran backups followed by `restic forget` for several months before noticing that it was eating a ton of diskspace. They I tried running `restic prune` to reclaim the space, but it failed

The repository is on another machine as accessed via NFS.  This failure is running the prune over NFS. If I run it directly on the server it runs out of memory very quickly. (very whimpy machine)

## Actual behavior

```
$ restic prune
counting files in repo
building new index for repo
[28:47] 100.00%  332112 / 332112 packs
repository contains 332112 packs (5251506 blobs) with 1.499 TiB bytes
processed 13790865 blobs: 8539359 duplicate blobs, 891.007 GiB duplicate
load all snapshots
find data that is still in use for 32 snapshots
[1:11:45] 100.00%  32 / 32 snapshots
found 2704802 of 13790865 data blobs still in use, removing 11086063 blobs
will delete 51619 packs and rewrite 271265 packs, this frees 284.798 GiB
header is larger than file
restic/pack.readHeader
        /tmp/restic-build-068254688/src/restic/pack/pack.go:201
restic/pack.List
        /tmp/restic-build-068254688/src/restic/pack/pack.go:223
restic/repository.Repack
        /tmp/restic-build-068254688/src/restic/repository/repack.go:38
main.runPrune
        /tmp/restic-build-068254688/src/cmds/restic/cmd_prune.go:211
main.glob..func11
        /tmp/restic-build-068254688/src/cmds/restic/cmd_prune.go:26
github.com/spf13/cobra.(*Command).execute
        /tmp/restic-build-068254688/src/github.com/spf13/cobra/command.go:599
github.com/spf13/cobra.(*Command).ExecuteC
        /tmp/restic-build-068254688/src/github.com/spf13/cobra/command.go:689
github.com/spf13/cobra.(*Command).Execute
        /tmp/restic-build-068254688/src/github.com/spf13/cobra/command.go:648
main.main
        /tmp/restic-build-068254688/src/cmds/restic/main.go:40
runtime.main
        /usr/local/go/src/runtime/proc.go:183
runtime.goexit
        /usr/local/go/src/runtime/asm_amd64.s:2086
```

## Steps to reproduce the behavior

Rerunning has the same failure.

**Comment by wscott** [comment]
On other observation. This prune seems to eat a bunch of diskspace and doesn't clean it all up at the end. Perhaps it ran out of diskspace and I should go look from tmp files.   No, don't see much.  Perhaps it is creating the new packs in the 'data' section and those are left dangling when it crashes.

**Comment by fd0** [comment]
Hey, thanks for the report. I suspect that it is basically a duplicate of #720. The `prune` command works by copying still needed data to new files in the repo, and only afterwards removing the old (now unused) files. That's a conservative approach. In this case, a bug occurs which makes restic panic, so it aborts and does not remove the unneeded files. That's consistent with your observation that the repo grows.

Could you run the following commands to see if there is any data corruption:
```
find /path/to/repo f | while read file; do echo "$(basename ${file}) ${file}"; done | tee sha256sums
sha256sums --quiet --check sha256sums
```

**Comment by fd0** [comment]
Hey, did you find time to run the check?

**Comment by wscott** [comment]
Sorry, but I kinda messed up my repository and I didn't want to bother you.
It was on a Synology NAS device and the repository had grown to over a TB and could only be accessed over the network.  (The synology cpu/memory are completely unable to run restic on a repository of that size.)

I _think_ the check didn't find anything, but I got the bright idea of just deleting chunks older than a certain date figuring a 'check' would notice the missing data and re-transfer it and notify me of any snapshots that can't be recovered. That wasn't the case at all and everything kinda died at that point.  So I deleted everything and started over.

This problem is caused because I read the 'forget' documentation and didn't realize that it doesn't actually delete any data.  Also, the performance of prune is such that it can't be run as part of the nightly backups.  These all make using this software a lot more complicated than I wanted.
